### PR TITLE
chore: cleanup table component eslint errors and warnings [OTE-290]

### PR DESCRIPTION
**Pull request by yogurtandjam**
cleans up eslint issues for the table component.

https://linear.app/dydx/issue/OTE-290/fix-eslint-srccomponents-table

### CODE REVIEW

File: src/components/Table.tsx [hunk 1]
```diff
@@ -77,17 +82,19 @@ export type ColumnDef<TableRowData extends object> = {
   width?: ColumnSize;
 };
 
-export type ElementProps<TableRowData extends object | CustomRowConfig, TableRowKey extends Key> = {
```
Comment (yogurtandjam) [inline src/components/Table.tsx:80]: extra generics was making this type more difficult to parse and we were never injecting unique generics that added any sort of type specificity.

the Key was always being implicitly typed and via the passed in props which would happen regardless if we added a generic for it or not.

File: .eslintignore [hunk 2]
```diff
@@ -5,11 +5,16 @@ vite.config.ts
...
 /src/main.tsx
 /src/menus
 /src/pages
 /src/views
+
+# components dir is being worked on
+/src/components/*
+
+# unignored components
+!/src/components/Table.tsx
```
Comment (jaredvu) [inline .eslintignore:20]: 🔥 genius

File: src/components/Table.tsx [hunk 3]
```diff
@@ -166,15 +173,15 @@ export const Table = <TableRowData extends object, TableRowKey extends Key>({
...
     const first = column?.getCellValue(a);
     const second = column?.getCellValue(b);
 
     return (
       // Compare the items by the sorted column
-      (isNaN(first as number)
+      (Number.isNaN(Number(first))
         ? // String
-          collator.compare(first as string, second as string)
+          collator.compare(String(first), String(second))
```
Comment (jaredvu) [inline src/components/Table.tsx:184]: i feel like we could use `typeof first === 'string' && typeof second === 'string'` instead of typescasting here

Comment (yogurtandjam) [reply src/components/Table.tsx:184]: i think it depends on the intended effect - do we want to be able to sort `'1'` with `2`? that's the current functionality which i'm preserving. (this should never happen though?)

i do agree your way is more correct though. how do you feel about doing that as a followup to this PR? so we can isolate that PR for regression tracking

**Review by jaredvu — APPROVED** [review]

**Pull request merged**